src/origin-matcher.js

```diff
@@ -1,4 +1,8 @@
 
+exports.generic = function (list) {
+  return list.length === 1 && list[0] === '*'
```

Comment:
I agree that's quite strange, not 100% happy with that design. Do you think it would make more sense to have some sort of option flag like

```js
{
  // specify one or more origins, wildcard supported
  // will only return the matched origin, with a Vary header
  origins: ['https://myapp.com', 'https://*.myapp.com']

  // or use the generic mode
  // which always returns "Accept", and can be cached across the board
  // but has security implications
  allowAllOrigins: true
}
```